File: 6kyu_FindTheParityOutlier.cs
using System.Collections.Generic;
using System.Linq;
using System;

public class Kata
{
  public static int Find(int[] integers)
  {
    var even_s = integers.Where(integ => integ % 2 == 0).ToList();
    var odd_s = integers.Where(integ => integ % 2 == 1).ToList();
    if (even_s.Count == 1)
    {
      return even_s[0];
    }
    else
    {
      return odd_s[0];
    } 
  }
}

File: 6kyu_FindTheParityOutlier.js
// https://www.codewars.com/kata/5526fc09a1bbd946250002dc/solutions/javascript

function findOutlier(integers){
  return (integers[0] % 2 !== 0 && integers[1] % 2 === 0 && integers[2] % 2 === 0) || 
    (integers[0] % 2 === 0 && integers[1] % 2 !== 0 && integers[2] % 2 === 0) || 
    (integers[0] % 2 === 0 && integers[1] % 2 === 0 && integers[2] % 2 !== 0) || 
    (integers[0] % 2 === 0 && integers[1] % 2 === 0 && integers[2] % 2 === 0) 
    ? integers.filter(integ => integ % 2 !== 0).pop() : integers.filter(integ => integ % 2 === 0).pop()
}

// function findOutlier(integers) {
//   const even_s = integers.filter(integ => integ % 2 === 0)
//   const odd_s = integers.filter(integ => (integ % 2 === 1 || integ % 2 === -1))
//   const result = (even_s.length === 1) ? even_s[0]: odd_s[0]
//   return result
// }
